uncollapsed_bam_processing.cwl

```diff
@@ -0,0 +1,750 @@
+class: Workflow
+cwlVersion: v1.0
+id: uncollapsed_bam_generation
+doc: >-
+  This is the workflow is written using Common Workflow Language (CWL) version
+  1.0 (https://www.commonwl.org/v1.0/) and is used at Memorial Sloan Kettering
+  Cancer Center for producing standard bam files from the NY state-approved
+  MSK-ACCESS assay. It is meant to be run on a single sample paired-end read 1
+  and read 2 fastq's, from Illumina sequencing data, but may be generalizable to
+  other sequencing platforms and sequencing panels as well, which produce
+  paired-end data.
+label: Uncollapsed BAM Generation
+$namespaces:
+  s: 'https://schema.org/'
+  sbg: 'https://www.sevenbridges.com/'
+inputs:
+  - id: input_fgbio_fastq_to_bam
+    type: 'File[]'
+    doc: Input Fastq.gz files for BAM generation
+    'sbg:x': 0
+    'sbg:y': 2033
+  - id: fgbio_fastq_to_bam_umi-tag
+    type: string?
+    'sbg:x': 0
+    'sbg:y': 2889
+  - id: fgbio_fastq_to_bam_sort
+    type: boolean?
+    'sbg:x': 0
+    'sbg:y': 2996
+  - id: sequencing-center
+    type: string?
+    'sbg:x': 0
+    'sbg:y': 321
+  - id: sample
+    type: string?
+    'sbg:x': 0
+    'sbg:y': 428
+  - id: run-date
+    type: string?
+    'sbg:x': 0
+    'sbg:y': 535
+  - id: read-structures
+    type: 'string[]?'
+    'sbg:x': 0
+    'sbg:y': 749
+  - id: read-group-id
+    type: string?
+    'sbg:x': 0
+    'sbg:y': 856
+  - id: fgbio_fastq_to_bam_predicted-insert-size
+    type: int?
+    'sbg:x': 0
+    'sbg:y': 3103
+  - id: platform-unit
+    type: string?
+    'sbg:x': 0
+    'sbg:y': 1284
+  - id: platform-model
+    type: string?
+    'sbg:x': 0
+    'sbg:y': 1391
+  - id: platform
+    type: string?
+    'sbg:x': 0
+    'sbg:y': 1498
+  - id: fgbio_fastq_to_bam_output_file_name
+    type: string?
+    'sbg:x': 0
+    'sbg:y': 3210
+  - id: library
+    type: string?
+    'sbg:x': 0
+    'sbg:y': 1926
+  - id: description
+    type: string?
+    'sbg:x': 0
+    'sbg:y': 4387
+  - id: comment
+    type: string?
+    'sbg:x': 0
+    'sbg:y': 4494
+  - id: validation_stringency
+    type: string?
+    'sbg:x': 0
+    'sbg:y': 0
+  - id: unpaired_fastq_file
+    type: string?
+    doc: Name of the Unpaired Fastq File
+    'sbg:x': 0
+    'sbg:y': 107
+  - id: gatk_sam_to_fastq_include_non_primary_alignments
+    type: boolean?
+    'sbg:x': 0
+    'sbg:y': 2140
+  - id: gatk_sam_to_fastq_include_non_pf_reads
+    type: boolean?
+    'sbg:x': 0
+    'sbg:y': 2247
+  - id: R1_output_fastq
+    type: string?
+    doc: Name of the R1 output Fastq File
+    'sbg:x': 0
+    'sbg:y': 1177
+  - id: R2_output_fastq
+    type: string?
+    doc: Name of the R2 Fastq File
+    'sbg:x': 0
+    'sbg:y': 1070
+  - id: reference_sequence
+    type: File
+    secondaryFiles:
+      - .amb
+      - .fai
+      - .sa
+      - .dict
+      - .ann
+      - ' .bwt'
```

Comment:
is the space supposed to be present here? or should it be `.bwt`?

Reply:
I'm also not used to all of these secondary files, I believe only the `.fai` file is required for BWA, but maybe I'm mistaken?